src/linear_algebra/affine_space/affine_subspace.lean

```diff
@@ -378,6 +378,23 @@ begin
     exact vsub_mem_direction hp hq2 }
 end
 
+instance to_add_torsor (s : affine_subspace k P) [nonempty s] : add_torsor s.direction s :=
+{ vadd := λ a b, ⟨(a:V) +ᵥ (b:P), vadd_mem_of_mem_direction a.2 b.2⟩,
+  zero_vadd' := by simp,
+  vadd_assoc' := λ a b c, by { ext, apply add_torsor.vadd_assoc' },
+  vsub := λ a b, ⟨(a:P) -ᵥ (b:P), (vsub_left_mem_direction_iff_mem a.2 _).mpr b.2 ⟩,
+  nonempty := by apply_instance,
+  vsub_vadd' := λ a b, by { ext, apply add_torsor.vsub_vadd' },
+  vadd_vsub' := λ a b, by { ext, apply add_torsor.vadd_vsub' } }
+
+@[simp] lemma coe_vsub (s : affine_subspace k P) [nonempty s] (a b : s) :
+  ↑(a -ᵥ b) = (a:P) -ᵥ (b:P) :=
+rfl
+
+@[simp] lemma coe_vadd (s : affine_subspace k P) [nonempty s] (a : s.direction) (b : s) :
+  ↑(a +ᵥ b) = (a:V) +ᵥ (b:P) :=
+rfl
```

Comment:
Should these be marked as `norm_cast` lemmas, or is there some reason that wouldn't work?

Reply:
Makes sense, I've made the change, but I'll leave the comment unresolved so someone more expert can confirm that it will work.

Reply:
@hrmacbeth You can test this very simply, sticking right after the lemma:
```lean
example (s : affine_subspace k P) [nonempty s] (a b : s) :
  ↑(a -ᵥ b) = (a:P) -ᵥ (b:P) := by norm_cast
```
(and checking that removing the `norm_cast` attribute on the lemma makes it fail).